enacts/wat_bal/maproom_monit.py

```diff
@@ -127,6 +127,9 @@ def update_time_sel(planting_day, planting_month, graph_click, current_options):
     if dash.ctx.triggered_id == "wat_bal_plot":
         time_range = current_options
         the_value = graph_click["points"][0]["x"]
+        for a_date in time_range:
+            if a_date.startswith(the_value):
```

Comment:
Looks like you're doing string operations on string representations of dates. That feels kind of icky--I probably would have parsed the dates to datetimes and then done date operations on them. I guess this way is fine if we're sure that the date strings are generated the same way, e.g. we're not going to have January represented as `1` in one place and `01` in another.

Reply:
So the difficulty here resides in the fact that we are plotting two time series, for the same days of different years. Because of that, we decided to remove the year from the label of the x-axis and so the_value is of the format d Mmm (e.g. 9 Oct) while time_range has the current year. So that's why I had to do this gymnastic to get the correct value in time_range list by having only the d Mmm from clicking the time series.

This being said, I preserved the year in the time series customdata but only for the current year so that it shows when mousing over the curve. Thus, I could retrieve the year from customdata. However, the user can also click the comparison year to select a date from time_range so I'd have to put the current years in the customdata of the comparison year and make sure it is not shows in the ts graphs...

Then once I have both day, month and year as strings from the click on the graph, I can indeed work on datetimes rather than strings. But all the above sounds tedious so maybe the whole paragdigme of doing this should be revisited. On the other hand, I do like that feature of clicking the time series to see the map at that time location and I think it then deserves to become part of general functionality to be saved somewhere... so maybe leave all that to a dedicated PR?

Reply:
Use datetimes as values, and hovertemplate and tickformat to define how they should be presented visually.

Reply:
This should be implemented in full now.

Reply:
This is what made us use %Y-%m-%d as a "standard" value to pass around callbacks, menus, functions and plots, and then reformat if we wanted to. I think part of the problem was discussed in the office so doesn't transpire here. I believe we could not use datetimes values because in the menu, it becomes a string or something. So we reverted to using %Y-%m-%d. I might be wrong... I don't remember well...

Reply:
I think dash uses JSON as its data representation, and JSON has no date type.